ENVParser/EnvFile.cs

```diff
@@ -0,0 +1,580 @@
+            DOF_GaussType = reader.ReadUInt32();
+            EnableSSAO = reader.ReadBoolean();
+            SSAO_OccluderRadius = reader.ReadSingle();
+            SSAO_FallOffRadius = reader.ReadSingle();
+            SSAO_BlurScale = reader.ReadSingle();
+            SSAO_Brightness = reader.ReadSingle();
+            SSAO_DepthRange = reader.ReadSingle();
+            DisableUnknownFlaggedSection = reader.ReadBoolean();
+            FieldShadowFarClip = reader.ReadSingle();
+            Field294 = reader.ReadSingle();
+            AmbientShadowBrightness = reader.ReadSingle();
+            Field29C = reader.ReadSingle();
+            Field2A0 = reader.ReadUInt32();
+            FieldShadowNearClip = reader.ReadSingle();
+            FieldShadowBrightness = reader.ReadSingle();
+            Field2AC = reader.ReadBoolean();
+            Field2AD = reader.ReadBoolean();
+            Field2AE = reader.ReadBoolean();
+            Field2AF = reader.ReadBoolean();
+            ShadowColourRed = reader.ReadSingle();
+            ShadowColourGreen = reader.ReadSingle();
+            ShadowColourBlue = reader.ReadSingle();
+            ShadowColourAlpha = reader.ReadSingle();
+            DisplayColourGrading = reader.ReadBoolean();
+            Cyan = reader.ReadSingle();
+            Magenta = reader.ReadSingle();
+            Yellow = reader.ReadSingle();
+            Dodge = reader.ReadSingle();
+            Burn = reader.ReadSingle();
+            LightMapR = reader.ReadSingle();
+            LightMapG = reader.ReadSingle();
+            LightMapB = reader.ReadSingle();
+            LightMapA = reader.ReadSingle();
+            EnableOutline = reader.ReadBoolean();
+            OutlineOpacity = reader.ReadSingle();
+            OutlineWidth = reader.ReadSingle();
+            CharacterOutlineBrightness = reader.ReadSingle();
+            Field2F2 = reader.ReadSingle();
+            Field2F6 = reader.ReadSingle();
+            ReflectionHeight = reader.ReadSingle();
+            EnablePhysicsSection = reader.ReadBoolean();
+            Gravity = reader.ReadSingle();
+            EnableWind = reader.ReadBoolean();
+            WindDirectionX = reader.ReadSingle();
+            WindDirectionY = reader.ReadSingle();
+            WindDirectionZ = reader.ReadSingle();
+            WindStrength = reader.ReadSingle();
+            WindStrengthModifier = reader.ReadSingle();
+            WindCycleTime = reader.ReadSingle();
+            WindCycleDelay = reader.ReadSingle();
+            ClearColourRed = reader.ReadByte();
+            ClearColourGreen = reader.ReadByte();
+            ClearColourBlue = reader.ReadByte();
+            ClearColourAlpha = reader.ReadByte();
+            Field324 = reader.ReadUInt32();
+            Field328 = reader.ReadUInt32();
+            Field32C = reader.ReadUInt32();
+            Field330 = reader.ReadUInt32();
+            Field334 = reader.ReadUInt32();
+            Field338 = reader.ReadByte();
```

Comment:
Just checking -- these various generically-named fields, `Field###`. Are they unknown, or padding? Like, are they always zero? Or do they vary in value?

Reply:
They are constant in every ENV. 

https://amicitia.miraheze.org/wiki/Persona_5_Royal/ENV/Structure#ENV_Footer - I've documented it here. Again can pull it out into its own footer class, but I'm really wary of creating my own types - maybe I could chuck it into a byte array instead, rather than splitting it up arbitrarily in these fields. (Though as you'll see on the wiki page, I split them because its not just 0 padding, there are values in there, which are constant across the ENVs)